pubtools/exodus/_tasks/push.py

```diff
@@ -0,0 +1,127 @@
+import logging
+import subprocess
+import time
+
+from six.moves.urllib.parse import urljoin
+
+from pubtools.exodus.task import ExodusTask
+
+LOG = logging.getLogger("pubtools-exodus")
+LOG_FORMAT = "%(asctime)s [%(levelname)-8s] %(message)s"
+
+
+class ExodusPushTask(ExodusTask):
+    """Push a directory to the Exodus CDN"""
+
+    def __init__(self):
+        super(ExodusPushTask, self).__init__()
+        self.publish = None
+        self.src = None
+        self.dest = None
+        self.prefix = None
+
+    def add_args(self):
+        super(ExodusPushTask, self).add_args()
+
+        self.parser.add_argument(
+            "--source",
+            help="The path to the directory to be pushed to the exodus CDN",
+            nargs="+",
+        )
+        self.parser.add_argument(
+            "--dest",
+            help="The destination path to publish to on the exodus CDN",
+        )
+        self.parser.add_argument(
```

Comment:
I'm questioning whether we need both of these.  They're just joined together later on, right?  It seems only a 'dest' would already be enough.

Correct me if I'm wrong but there's no difference between e.g. "--dest /foo/bar --prefix exodus:/quux" and "--dest /bar --prefix exodus:/quux/foo".  It seems confusing to me.  As a caller, if I know where I'm publishing then I can arbitrarily split that between "prefix" and "dest" at any point within the string.  So why wouldn't I always stuff the entire string into one of these arguments and ignore the other.  What am I supposed to put in dest, what am I supposed to put in prefix and why?

Also, do you have a clear idea yet how exodus-rsync.conf is going to look on the hosts where this is run?  For example would there be an "exodus-qa", "exodus-stage" etc prefix, or would there just be an "exodus" which relies on the existing  env vars like EXODUS_GW_URL ?

If it's the latter then may I suggest implementing default behavior of the form, "if the destination doesn't contain ':' it is prefixed with 'exodus:'" so that one doesn't need to know or care what's in exodus-rsync.conf to use this tool in the common case.